### PR TITLE
bpo-41845: Move PyObject_GenericGetDict() back into the limited API

### PR DESCRIPTION
**Pull request by ZackerySpytz**
It was moved out of the limited API in
7d95e4072169911b228c9e42367afb5f17fd3db0.

https://bugs.python.org/issue41845

### CODE REVIEW

**Review by alex — COMMENTED** [review]
You need to add this to `PC/python3dll.c` as well

**Review by alex — APPROVED** [review]
LGTM, but waiting for a windows reviewer.

File: Include/dictobject.h [hunk 1]
```diff
@@ -57,6 +57,9 @@ PyAPI_FUNC(int) PyDict_MergeFromSeq2(PyObject *d,
 PyAPI_FUNC(PyObject *) PyDict_GetItemString(PyObject *dp, const char *key);
 PyAPI_FUNC(int) PyDict_SetItemString(PyObject *dp, const char *key, PyObject *item);
 PyAPI_FUNC(int) PyDict_DelItemString(PyObject *dp, const char *key);
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x03030000
```
Comment (zooba) [inline Include/dictobject.h]:
```suggestion
#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x030A0000
```

Unfortunately, we can't backport this change, so we need to set the minimum version to 3.10.

File: Misc/NEWS.d/next/C API/2020-10-11-05-05-53.bpo-41845.ZFvuQM.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+:c:func:`PyObject_GenericGetDict` is available again in the limited API.
```
Comment (zooba) [inline Misc/NEWS.d/next/C API/2020-10-11-05-05-53.bpo-41845.ZFvuQM.rst]:
```suggestion
:c:func:`PyObject_GenericGetDict` is available again in the limited API when targeted 3.10 or later.
```

**Review by zooba — CHANGES_REQUESTED** [review]

**Comment by ZackerySpytz** [comment]
I have made the requested changes; please review again.

**Review by zooba — APPROVED** [review]

**Review by alex — APPROVED** [review]

**Pull request merged**